src/components/AccordionPanel/AccordionPanel.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useState } from 'react'
+
+import { mapToArray, arrayToMap } from './AccordionPanelHelper'
+
+type Props = {
+  children: React.ReactNode
+  icon?: 'left' | 'right' | 'none'
+  expandableMultiply?: boolean
+  defaultExpanded?: string[]
+  className?: string
+  onClick?: (expandedItems: string[]) => void
+}
+
+export const AccordionPanelContext = React.createContext<any>({
+  expandedItems: '',
+  icon: 'left',
+  onClick: () => {},
+})
+
+export const AccordionPanel: React.FC<Props> = ({
+  children,
+  icon = 'left',
+  expandableMultiply = false,
+  defaultExpanded = [],
+  className = '',
+  onClick,
+}) => {
+  const [expandedItems, setExpanded] = useState(new Map(arrayToMap(defaultExpanded)))
+
+  const handleClick = useCallback(
+    (itemName: string, isExpanded: boolean) => {
+      if (expandableMultiply) {
+        isExpanded ? expandedItems.set(itemName, itemName) : expandedItems.delete(itemName)
+        setExpanded(new Map(expandedItems))
```

Comment:
[MUST] Rewriting value directly for `expandedItems` state. Instead of creating a new Map when executing setExpanded, execute set or delete on the newly created Map from expandedItems.